mne_bids/meg_bids.py

```diff
@@ -267,99 +246,110 @@ def raw_to_bids(subject_id, session_id, run, task, raw_fname, output_path,
-                                  head_shape_fname=hsp,
-                                  preload=False, verbose=verbose)
-
-    # CTF systems
-    elif ext == '.ds':
-        raw = io.read_raw_ctf(raw_fname)
+    if isinstance(raw_file, str):
+        # We must read in the raw data
+        raw = _read_raw(raw_file, electrode=electrode, hsp=hsp, hpi=hpi,
+                        config=config, verbose=verbose)
+        _, ext = _parse_ext(raw_file)
+    elif isinstance(raw_file, BaseRaw):
+        # Only parse the filename for the extension
+        # Assume that if no filename attr exists, it's a fif file.
+        raw = raw_file
+        if hasattr(raw, 'filenames'):
+            _, ext = _parse_ext(raw.filenames[0])
+        else:
+            ext = '.fif'
+    data_path = make_bids_folders(subject=subject_id, session=session_id,
+                                  kind=kind, root=output_path)
+    if session_id is None:
+        ses_path = data_path
+    else:
+        ses_path = make_bids_folders(subject=subject_id, session=session_id,
+                                     root=output_path)
+
+    # create filenames
+    scans_fname = make_bids_filename(
+        subject=subject_id, session=session_id, suffix='scans.tsv',
+        prefix=ses_path)
+
+    coordsystem_fname = make_bids_filename(
+        subject=subject_id, session=session_id,
+        suffix='coordsystem.json', prefix=data_path)
+    data_meta_fname = make_bids_filename(
+        subject=subject_id, session=session_id,
+        suffix='%s.json' % kind, prefix=data_path)
+    raw_file_bids = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        suffix='%s%s' % (kind, ext), prefix=data_path)
+    events_tsv_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task,
+        run=run, suffix='events.tsv', prefix=data_path)
+    channels_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        suffix='channels.tsv', prefix=data_path)
+
+    # Read in Raw object and extract metadata from Raw object if needed
+    if kind == 'meg':
+        orient = orientation[ext]
+        unit = units[ext]
+        manufacturer = manufacturers[ext]
+    else:
+        orient = None
+        unit = None
+        manufacturer = None
 
     # save stuff
-    _dataset_description_json(output_path, verbose)
-    _scans_tsv(raw, raw_fname_bids, scans_fname, verbose)
-    _coordsystem_json(raw, unit, orient, manufacturer, fid_fname, verbose)
-    _meg_json(raw, task, manufacturer, meg_fname, verbose)
+    if kind == 'meg':
+        _scans_tsv(raw, raw_file_bids, scans_fname, verbose)
+        _coordsystem_json(raw, unit, orient, manufacturer, coordsystem_fname,
+                          verbose)
+
+    make_dataset_description(output_path, name=" ",
+                             bids_version="1.0.2 (draft)", verbose=verbose)
```

Comment:
should we make bids_version a global variable and hoist it up. I feel having defined in the function will leave to it being forgotten with updates to the codebase

Reply:
Agreed !